qiita_db/analysis.py

```diff
@@ -200,39 +198,39 @@ def delete(cls, _id):
         QiitaDBUnknownIDError
             If the analysis id doesn't exist
         """
-        with TRN:
+        with qdb.sql_connection.TRN:
```

Comment:
I suspect you may have been able to get away by doing `with qdb.sql_connection.TRN as TRN:`.